app/views/users/show.html.erb

```diff
@@ -1,16 +1,38 @@
 <h1>User Page with List of Posts</h1>
-<p>User 1</p>
-<p>Bio: Lorem ipsum dolor sit amet, consectetur adipiscing elit, 
+<div class="user-line">
+    <%= link_to image_tag(@user.photo, class: 'user-photo'), user_path(@user) %>
+    <div class="name-counter">
+      <span class="name">
+        <%= link_to @user.name, user_path(@user) %>
+      </span>
+      <div>
+        <span class="right-top"></span>
+        <span class="counter">Number of posts: <%= @user.posts_counter %></span>
+        <span class="right"></span>
+      </div>
+    </div>
+</div>
+<section>
+  <h4>Bio:</h4>
+  <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, 
   sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. 
   Ac tortor dignissim convallis aenean et. Lectus magna fringilla urna 
   porttitor rhoncus dolor. Consequat interdum varius sit amet mattis 
-  vulputate. Fringilla urna porttitor rhoncus dolor purus non enim. 
-  Pharetra massa massa ultricies mi quis hendrerit. Praesent elementum 
-  facilisis leo vel. Ultrices neque ornare aenean euismod elementum 
-  nisi quis eleifend quam. Turpis cursus in hac habitasse platea 
-  dictumst quisque sagittis.</p>
-<ul>
-  <li>Post 1</li>
-  <li>Post 2</li>
-  <li>Post 3</li>
-</ul>
+  vulputate. Fringilla urna porttitor rhoncus dolor purus non enim.</p>
+</section>
+
+<% @posts.each do |post| %>
+  <div class="post-border">
+    <p>Post <%= post.id %>: <%= post.title %></p>
+  </div>
+<% end %>
+
+<div class="sep"></div>
+
+<% if @show_all_posts_button %>
+  <%= link_to "See All Posts", user_posts_path(@user), class: "button" %>
+<% end %>
+
+<div class="sep"></div>
+
+<%= will_paginate @posts %>
```

Comment:
- [ ] 👍 Good progress on your project! However, based on the wireframe specifications, there's no requirement for pagination, and it's suggested to show 3 recent posts per user without pagination. I recommend removing the pagination feature to align with the wireframe and project specifications.

Keep up the good work! If you have any questions or need further assistance, feel free to reach out. 🚀👩‍💻